### PR TITLE
Create dropdown

### PR DESCRIPTION
**Pull request by BenjaminParment**
Im having some issues with how we should structure the dropdown because I tried to make it generic so it (just like the list, takes a array of objects type T, and we extract the string name using extract function within each object. However, when selecting an option in the dropdown, the only accepted types for the onChange function parameter is
-string
-number
-undefined

So i dont really know how to make it accept a type T object

### CODE REVIEW

**Comment by BenjaminParment** [comment]
I think option two is the way to go if we want to keep this class as generic as possible.

**Comment by j-n-white** [comment]
I agree

File: src/components/reusables/dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export interface DropdownRowDef<T> {
+    key: string;
+    extractContent: (data: T) => string;
+}
+
+interface DropdownProps<T> {
+    defaultOption: string;
+    data: T;
+    options: DropdownRowDef<T>[];
+    callbackFunction: (data: string) => void;
```
Comment (j-n-white) [inline src/components/reusables/dropdown/Dropdown.tsx]: If we are going down this rout then we don't need all these props. The options can just be a list of strings or a list of tuples of value and display.

File: src/components/reusables/dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export class Dropdown<T> extends React.Component<DropdownProps<T>> {
+    private onChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        const value = event.target.value;
+        this.props.callbackFunction(value);
+    };
+
+    render = () => (
+        <div className="dropdown-selector-paren-container">
+            <select onChange={this.onChange}>
+                <option value="">{this.props.defaultOption}</option>
```
Comment (j-n-white) [inline src/components/reusables/dropdown/Dropdown.tsx]: The default option is not always going to be the one at the top of the list so if you just make the options contain all the options and the select the correct on on initial render.

Comment (j-n-white) [reply src/components/reusables/dropdown/Dropdown.tsx]: We might also need a placeholder text, you'll need to check the designs.

File: src/components/reusables/dropdown/Dropdown.tsx (hunk 1 above)
Comment (BenjaminParment) [reply src/components/reusables/dropdown/Dropdown.tsx]: I will make it a list of strings then, but I think we should add a "key" props to pass to each options

Comment (j-n-white) [reply src/components/reusables/dropdown/Dropdown.tsx]: The key can be the string. If they are not all different anyway it is going to be confusing.

Comment (BenjaminParment) [reply src/components/reusables/dropdown/Dropdown.tsx]: the big bad dude from yesterday said a key cannot have any spaces, so what do we do if the value is "option 1", turn it into option1?

**Pull request merged**